### DISCUSSION TITLE
Parse nan and inf into numbers

### DISCUSSION THREAD

**Discussion by drhagenabm** [eclipse-langium/langium#1596] [Q&A]
Langium has the ability to parse strings into numbers so that not all attributes need to be strings. This also works on "NaN" and "Infinity". Is there a way to parse strings other than "NaN" and "Infinity" into numbers? I mean, `terminal NAN returns number: 'nan';` works for funny reasons, but `terminal INFINITY returns number: 'inf';` does not.

For example, this grammar parsers "inf" as `NaN`:

```langium
grammar HelloWorld

entry Model:
    (number+=FloatLiteral)*;

FloatLiteral:
    value=FLOAT;

terminal FINITE_FLOAT returns number: /\d+((\.\d+([Ee][+-]?\d+)?)|((\.\d+)?[Ee][+-]?\d+))/;
terminal NAN returns number: 'nan';
terminal INFINITY returns number: 'inf';
FLOAT returns number:
    NAN | INFINITY | FINITE_FLOAT;

hidden terminal WS: /\s+/;
terminal ID: /[_a-zA-Z][\w_]*/;

hidden terminal ML_COMMENT: /\/\*[\s\S]*?\*\//;
hidden terminal SL_COMMENT: /\/\/[^\n\r]*/;
```

**Comment by msujew** [comment]
Hey @drhagenabm,

the value conversion for terminals and data type rules happens in the [`ValueConverter`](https://github.com/eclipse-langium/langium/blob/main/packages/langium/src/parser/value-converter.ts). By default, it will just use `Number('input')` to convert to a number. You can override the service to adjust the behavior.
